activestorage/lib/active_storage/transformers/image_processing_transformer.rb

```diff
@@ -105,6 +105,12 @@ def validate_arg_hash(argument)
             end
           end
         end
+
+        def reformat(format)
+          return format unless ActiveStorage.variant_processor == :vips
+
+          { "jfif" => "jpg" }.fetch(format, format)
```

Comment:
this can be a constant or another private method for mappings.

there is already mini magick logic in this file as well, so wouldn't be that shocking.
```ruby
    if ActiveStorage.variant_processor == :mini_magick
              validate_transformation(name, argument)
      end
```